### PR TITLE
add codecov config

### PR DESCRIPTION
**Pull request by wilzbach**
an attempt to customize CodeCov.

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    patch:
+      default:
+        enabled: yes
+        if_no_uploads: error
+
+    changes: true
+
+
+comment:
+  layout: "header, diff, changes, tree, uncovered"
```
Comment (wilzbach) [inline .codecov.yml]: The tree layout is a moderately new feature from CodeCov - you can see a screenshot here https://github.com/codecov/support/issues/171

**Comment by wilzbach** [comment]
Similar as https://github.com/dlang/druntime/pull/1623 and https://github.com/dlang/dmd/pull/6027 this will disable the CodeCov bot for commenting. The status will still be displayed in the CI integration box (red cross if the coverage decreases) and interested reviewers might want to download the [Browser extension](https://github.com/codecov/browser-extension) to see missed lines highlighted directly on the diff.

CC @MartinNowak

**Comment by wilzbach** [comment]
To give codecov the permission to post to the CI integration board, I added `dlang-bot` to the config file.

See also:

https://github.com/dlang/dmd/pull/6046
https://github.com/dlang/druntime/pull/1628

**Comment by AndrejMitrovic** [comment]
Auto-merge toggled on

**Pull request merged**